include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -494,18 +530,40 @@ namespace internal
                     }
             }
         }
+
+      univariate_shape_data.element_type = this->element_type;
```

Comment:
I am wondering about this line. Is this actually necessary to have the information both in `ShapeData` and `UnivariateShapeData`? To me it looks like redundant information: we could probably replace the field in `ShapeData` by a reference and deprecate it (if we want to be backwards compatible) or through it out? Or is `ShapeData::element_type` meant like a cache and it stores the stores the worst case/upper bound?

Reply:
If I remember correctly I discussed this with Martin: the problem is that enum `ElementType` includes one-dimensional (univariate) as well as d-dimensional (at the element level) concepts. At the moment there might be no specific use case but I believe it is not too far-fetched to come up with one (maybe RT elements with anisotropic tensor structure).

Reply:
Thanks @jwitte08!